msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase1561136.java

```diff
@@ -0,0 +1,129 @@
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.MicrosoftStsPromptHandlerParameters;
+import com.microsoft.identity.labapi.utilities.client.ILabAccount;
+import com.microsoft.identity.labapi.utilities.client.LabQuery;
+import com.microsoft.identity.labapi.utilities.constants.AzureEnvironment;
+import com.microsoft.identity.labapi.utilities.constants.TempUserType;
+import org.junit.Assert;
+import org.junit.Test;
+import java.util.Arrays;
+import java.util.List;
+
+// Get Broker Accounts
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1561136
+@SupportedBrokers(brokers = BrokerHost.class)
+public class TestCase1561136 extends AbstractMsalBrokerTest {
+
+    @Test
+    public void test_1561136() throws Throwable {
+        final String username = mLabAccount.getUsername();
+        final String password = mLabAccount.getPassword();
+
+        BrokerHost brokerHost = (BrokerHost) mBroker;
+        // Get accounts without signing in, does not return any accounts
+        Assert.assertEquals(brokerHost.getAllAccounts(false).size(), 0);
+
+        // Make an interactive call with MSAL
+        final MsalSdk msalSdk = new MsalSdk();
+        final MsalAuthTestParams authTestParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(username)
+                .scopes(Arrays.asList(mScopes))
+                .promptParameter(Prompt.SELECT_ACCOUNT)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+        final MsalAuthResult authResult = msalSdk.acquireTokenInteractive(authTestParams, new OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                final MicrosoftStsPromptHandlerParameters promptHandlerParameters = MicrosoftStsPromptHandlerParameters.builder()
+                        .prompt(PromptParameter.SELECT_ACCOUNT)
+                        .loginHint(username)
+                        .sessionExpected(false)
+                        .consentPageExpected(false)
+                        .build();
+
+                new MicrosoftStsPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        }, TokenRequestTimeout.MEDIUM);
+
+        authResult.assertSuccess();
+
+        // Check get accounts returns the account signed in with MSAL
+        List<String> accounts = brokerHost.getAllAccounts(false);
+        Assert.assertEquals(accounts.size(), 1);
+
+        // create another temp user
+        final ILabAccount labAccount = mLabClient.createTempAccount(TempUserType.BASIC);
+        final String username2 = labAccount.getUsername();
+        final String password2 = labAccount.getPassword();
+
+        Assert.assertNotEquals(username, username2);
+        // user-based join
+        mBroker.performDeviceRegistration(username2, password2);
```

Comment:
test seems to fail at this step..it seems due to replication delay....ADRS doesn't seem to know about temp user

Reply:
I am able to run this successfully in my local. Let me add a sleep time here for this to always work